cmd/admin-handlers.go

```diff
@@ -1529,3 +1529,36 @@ func (a adminAPIHandlers) TraceHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+// SpeedTestHandler - POST /minio/admin/v1/speedtest
+// ----------------
+// The handler performs a write and read speed test on all the drives in MinIO setup
+func (a adminAPIHandlers) SpeedTestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "SpeedTest")
+
+	// Fetch storage disks from the objectAPI instance.
+	objectAPI := validateAdminReq(ctx, w, r)
```

Comment:
Since `objectAPI` is not used anywhere else, we can go with the standard GO way.

    `if objectAPI := validateAdminReq(ctx, w, r); objectAPI == nil {`